Import css/css-text-decor/animations tests from the latest WPT

shivamidow opened the pull request:
#### 39f8d3a5c26cd0ebe9ebc95710075184d540f011
<pre>
Import css/css-text-decor/animations tests from the latest WPT
<a href="https://bugs.webkit.org/show_bug.cgi?id=263241">https://bugs.webkit.org/show_bug.cgi?id=263241</a>

Reviewed by Tim Nguyen.

This change imports animation tests related to CSS Text Decoration from
upstream WPT. We will use them to verify the animation of
text-decoration-thickness in this PR [1].

WPT hash for this import: 3d084c0ac239eff15e62958a18b5508203dc2aaa

[1] <a href="https://github.com/WebKit/WebKit/pull/19085">https://github.com/WebKit/WebKit/pull/19085</a>

* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/animations/discrete-no-interpolation-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/animations/discrete-no-interpolation.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/animations/text-underline-offset-interpolation-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/animations/text-underline-offset-interpolation.html: Added.

Canonical link: <a href="https://commits.webkit.org/269447@main">https://commits.webkit.org/269447@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c357217853a42f22060045fbff959d1a97e1dda1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22508 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/48/builds/265 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23592 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24416 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20837 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/1345 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/23040 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/21830 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22748 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/176 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19540 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/25265 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/176 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/20397 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/26683 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/20551 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/20643 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24490 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/96 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/17955 "Passed tests") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/71 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/20201 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5378 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/116 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/111 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
Would be good to include WPT upstream commit in the message

shivamidow committed:
Import css/css-text-decor/animations tests from the latest WPT
https://bugs.webkit.org/show_bug.cgi?id=263241

Reviewed by Tim Nguyen.

This change imports animation tests related to CSS Text Decoration from
upstream WPT. We will use them to verify the animation of
text-decoration-thickness in this PR [1].

WPT hash for this import: 3d084c0ac239eff15e62958a18b5508203dc2aaa

[1] https://github.com/WebKit/WebKit/pull/19085

* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/animations/discrete-no-interpolation-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/animations/discrete-no-interpolation.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/animations/text-underline-offset-interpolation-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/animations/text-underline-offset-interpolation.html: Added.

Canonical link: https://commits.webkit.org/269447@main
